development/extensions/tutorial_php_events_and_extending_extensions.rst

```diff
@@ -0,0 +1,325 @@
+=========================================================
+Tutorial: Allowing other extensions extend your extension
+=========================================================
+
+Some parts of this tutorial are based on the work by `imkingdavid <https://www.phpbb.com/community/viewtopic.php?f=461&t=2210001>`_
+
+Introduction
+============
+
+This tutorial explains:
+
+* `Adding a PHP event to your extension`_
+* `Adding a template event to your extension`_
+* `Using service collections`_
+* `Using the phpBB finder tool`_
+* `Using service replacement`_
+* `Using service decoration`_
+* `Replacing the phpBB Datetime class`_
+
+Adding a PHP event to your extension
+====================================
+The first of the these is probably the easiest, but is limited in abilities once
+implemented. Basically, this involves you, the extension author, making your own
+events at key points in the code of your extension that other extensions can then
+listen for. If your extension is not enabled, the relevant code in other
+extensions will be ignored. When your extension is enabled and the event in
+question is called, the listeners in the other enabled extensions that are
+listening for that event will be run.
+
+This is what an event looks like:
+
+.. code-block:: php
+
+    <?php
+    /**
+    * Short description of the event
+    *
+    * @event acme.demo.identifier
+    * @var  string  var1  A variable to make available to the event
+    * @var  string  var2  A second variable to make available to the event
+    * @since 3.1-A1
+    */
+    $vars = array('var1', 'var2');
+    extract($phpbb_dispatcher->trigger_event('acme.demo.identifier', compact($vars)));
```

Comment:
I wonder if we should talk about the dispatch() method and the ability of using a custom class even here... maybe in a `note` block or something like that with a link to the related symfony documentation?